File: src/Application Specific/Classes/Entity Collections/Scheduler/HeartBeat.vb
Namespace Entity
    Namespace HeartBeats
        Public Class HeartBeat

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

            Private _LockedVisitID As Integer = 0
            Public ReadOnly Property LockedVisitID() As Integer
                Get
                    Return _LockedVisitID
                End Get
            End Property
            Public WriteOnly Property SetLockedVisitID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_LockedVisitID", Value)
                End Set
            End Property


            Private _LastHeartBeat As DateTime = Nothing
            Public Property LastHeartBeat() As DateTime
                Get
                    Return _LastHeartBeat
                End Get
                Set(ByVal Value As DateTime)
                    _LastHeartBeat = Value
                End Set
            End Property


            Private _LastCheck As DateTime = Nothing
            Public Property LastCheck() As DateTime
                Get
                    Return _LastCheck
                End Get
                Set(ByVal Value As DateTime)
                    _LastCheck = Value
                End Set
            End Property


        End Class
    End Namespace
End Namespace

File: src/Application Specific/Classes/Entity Collections/Scheduler/HeartBeat.cs
using System;

namespace FSM.Entity
{
    namespace HeartBeats
    {
        public class HeartBeat
        {
            private DataTypeValidator _dataTypeValidator;

            public HeartBeat()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            private int _LockedVisitID = 0;

            public int LockedVisitID
            {
                get
                {
                    return _LockedVisitID;
                }
            }

            public object SetLockedVisitID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_LockedVisitID", value);
                }
            }

            private DateTime _LastHeartBeat = default;

            public DateTime LastHeartBeat
            {
                get
                {
                    return _LastHeartBeat;
                }

                set
                {
                    _LastHeartBeat = value;
                }
            }

            private DateTime _LastCheck = default;

            public DateTime LastCheck
            {
                get
                {
                    return _LastCheck;
                }

                set
                {
                    _LastCheck = value;
                }
            }
        }
    }
}